mcs/class/System.ServiceModel/Test/System.ServiceModel.Dispatcher/Bug32886Test.cs

```diff
@@ -63,16 +63,21 @@ public class Bug32886
 				var client = new TempConvertSoapClient (binding, remoteAddress);
 
 				var wait = new ManualResetEvent (false);
+
+				Exception error = null;
+				string result = null;
+
 				client.CelsiusToFahrenheitCompleted += delegate (object o, CelsiusToFahrenheitCompletedEventArgs e) {
-					if (e.Error != null)
-						throw e.Error;
-					Assert.AreEqual ("76.1", e.Result, "#1");
+					error = e.Error;
+					result = e.Result;
 					wait.Set ();
 				};
 
 				client.CelsiusToFahrenheitAsync ("24.5");
-				if (!wait.WaitOne (TimeSpan.FromSeconds (20)))
-					Assert.Fail ("timeout");
+
+				Assert.IsTrue (wait.WaitOne (TimeSpan.FromSeconds (20)), "timeout");
+				Assert.AreEqual (null, error, "#1, inner exception: {0}", error.ToString ());
```

Comment:
We don't need `error.ToString()` here and in the other tests, NUnit already provides that in the default assert message.